Adding the corresponding tests for the main project: movies_bloc

HoracioDuca opened the pull request:
### Adding the corresponding tests for the main project: "The Movies App"

**Mockito**, **Test** and **BuildRunner** libraries were necessary and it were added by _dependencies_.

In this case I added 5 tests to verify the correct operation for the following widgets :

• **MoviesBloc**
• **MoviesApiProvider**
• **HomePage** (main screen)
• **MovieDetail** (secondary screen)
• **MovieSearchBar** 

Also a **buildWidget** was necessary to create to print the screens correctly.

![image](https://user-images.githubusercontent.com/84336551/126642757-1b6fade3-59ae-46d1-ac9d-2fe9bf998853.png)

HoracioDuca committed:
Adding the corresponding tests for the main project: movies_bloc

movies_bloc_unit_test was renamed to just movies_bloc_test to keep
project consistency

FranciscoGaraventa reviewed (approved):
LGTM

MarcosCGlobant commented on lib/src/ui/movie_detail.dart:
```diff
@@ -7,9 +7,9 @@ import '../models/result_movie.dart';
 
-  const MovieDetail({
+  const MovieDetail(
+    this.result, {
     Key? key,
-    required this.result,
```
Why did you made this change?

HoracioDuca replied:
At first I had an error so I had solved it that way, but now, checking, I think you're right and it looks better as it was before, so I went back this change.

FranciscoGaraventa commented on lib/src/ui/movie_detail.dart:
```diff
@@ -7,9 +7,9 @@ import '../models/result_movie.dart';
 class MovieDetail extends StatelessWidget {
   final ResultMovie result;
```
In this case instead of using all the result object you can use it only the parameters that you need. You only need: posterPath, title, releaseDate, voteAverage, overview, originalLanguage, voteCount

HoracioDuca replied:
Done, movie_detail file now receive that parameters from movie_grid_builder instead of passing all the ResultMovie object. Also the movie_detail_test file needed to be modified.

HoracioDuca committed:
movie_detail was modified to receive only the needed parameters and movie_detail_test was adjusted

FranciscoGaraventa commented on lib/src/ui/movie_detail.dart:
```diff
@@ -36,22 +47,21 @@ class MovieDetail extends StatelessWidget {
-                      movieVoteCount: result.voteCount.toString(),
+                      movieReleaseDate: releaseDate,
+                      movieOriginalLanguage: originalLanguage.toUpperCase(),
+                      movieVoteCount: voteCount.toString(),
```
If you use this as String, avoid to convert inside the widget, change voteCount to String, the same to originalLanguage, pass it converted to uppercase

HoracioDuca replied:
Done

HoracioDuca committed:
voteCount changed from int to String and originalLanguage is now converted to upperCase from the result_movie model

originalLanguage is now converted toUpperCase when MovieDetail is
created

result_movie is now correctly structured and working fine

FranciscoGaraventa commented on lib/src/models/result_movie.dart:
```diff
@@ -55,8 +55,8 @@ class ResultMovie {
           : MovieStrings.defaultImage,
-      originalLanguage:
-          parsedJson['original_language'] ?? MovieStrings.languageNotAssigned,
+      originalLanguage: parsedJson['original_language'].toUpperCase() ??
```
Here is not necessary to do de toUpperCase(),you need to convert to upperCase when you are going to use it i mean that when you create MovieDetail( ... , originalLanguage: snapshot.data.results[index].originalLanguage.toUpperCase(), ... ) that is a correct implementation

HoracioDuca replied:
Fixed! Thank you

MarcosCGlobant reviewed (approved):
LGTM
